### ISSUE TITLE
Neither single stepping nor exception reporting show within blocks whose editor is open

### ISSUE THREAD

**Issue by kephas** [jmoenig/Snap#2311]
When an exception occurs with a custom block, even if its editor is open, there is no way to tell where the error occured. Similarily (but maybe it should be a separate issue), single stepping will only show execution of blocks in the script area, which makes it way less useful when using custom blocks.

**Comment by brianharvey** [comment]
The second point isn't true.  If you want to visible step inside a block, turn on visible stepping, _then_ open a Block Editor for that block.  It opens a special variant editor that does support stepping.

But I agree about the first point.  We are going to take a serious look at debugging support When Things Slow Down™.  Meanwhile Jens doesn't want to invent stopgap kludges.

**Comment by kephas** [comment]
This is a hugely useful feature, shouldn't that be way more visible? Best case, when you turn visible stepping on or off, the editors get switched.

**Comment by cycomachead** [comment]
> But I agree about the first point. We are going to take a serious look at debugging support When Things Slow Down™. Meanwhile Jens doesn't want to invent stopgap kludges.

Can we please have a discussion on this sooner than later? #1781 is a start. It seems better than "inside" without being in your face.

**Comment by brianharvey** [comment]
I agree.  It's currently showing merge conflicts.  Can you fix those?
